core/src/apps/common/address_mac.py

```diff
@@ -0,0 +1,32 @@
+from typing import TYPE_CHECKING
+
+from trezor import utils, wire
+from trezor.crypto import hashlib, hmac
+
+from .writers import write_bytes_unchecked, write_compact_size, write_uint32_le
+
+if TYPE_CHECKING:
+    from apps.common.coininfo import CoinInfo
+    from apps.common.keychain import Keychain
+
+_ADDRESS_MAC_KEY_PATH = [b"SLIP-0024", b"Address MAC key"]
+
+
+def check_address_mac(
+    address: bytes, mac: bytes, coin: CoinInfo, keychain: Keychain
+) -> None:
+    expected_mac = get_address_mac(address, coin, keychain)
+    if len(mac) != hashlib.sha256.digest_size or not utils.consteq(expected_mac, mac):
+        raise wire.DataError("Invalid address MAC.")
+
+
+def get_address_mac(address: bytes, coin: CoinInfo, keychain: Keychain) -> bytes:
```

Comment:
https://github.com/trezor/trezor-firmware/pull/2069/commits/1b42dd3abf091de4a0a81071e4cca24a3a239754 (note this will cause device tests to fail until further fixes are added)